docs/docs/03-light-clients/04-wasm/05-governance.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Governance
+sidebar_label: Governance
+sidebar_position: 5
+slug: /ibc/light-clients/wasm/governance
+---
+
+# Governance
+
+Learn how to upload Wasm light client byte code on a chain, and how to migrate an existing Wasm light client contract. {synopsis}
+
+## Setting an authority
+
+Both the storage of Wasm light client byte code as well as the migration of an existing Wasm light client contract are permissioned (i.e. only allowed to an authority such as governance). The designated authority is specified when instantiating `08-wasm`'s keeper: both [`NewKeeperWithVM`](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L33-L38) and [`NewKeeperWithConfig`](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L52-L57) constructor functions accept an `authority` argument that must be the address of the authorized actor. For example, in `app.go`, when instantiating the keeper, you can pass the address of the governance module:
+
+```go
+// app.go
+import (
+  ...
+  "github.com/cosmos/cosmos-sdk/runtime"
+  authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+  govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+  wasmkeeper "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/keeper"
+  wasmtypes "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+  ...
+)
+
+// app.go
+app.WasmClientKeeper = wasmkeeper.NewKeeperWithVM(
+  appCodec,
+  runtime.NewKVStoreService(keys[wasmtypes.StoreKey]),
+  app.IBCKeeper.ClientKeeper,
+ 	authtypes.NewModuleAddress(govtypes.ModuleName).String(), // authority
+  wasmVM,
+)
+```
+
+## Storing new Wasm light client byte code
+
+ If governance is the allowed authority, the governance v1 proposal that needs to be submitted to upload a new light client contract should contain the message [`MsgStoreCode`](https://github.com/cosmos/ibc-go/blob/f822b4fa7932a657420aba219c563e06c4465221/proto/ibc/lightclients/wasm/v1/tx.proto#L16-L23) with the base64-encoded byte code of the Wasm contract. Use the following CLI command and JSON as an example:
+
+```shell
+%s tx gov submit-proposal <path/to/proposal.json> --from <key_or_address>
```

Comment:
I feel like showing `simd tx gov ...` is a bit more readable than the `%s tx gov` especially since the later section showing the full commands uses `simd`